core/http-auth-aws/src/main/java/software/amazon/awssdk/http/auth/aws/crt/internal/signer/DefaultAwsCrtV4aHttpSigner.java

```diff
@@ -205,20 +208,22 @@ private static void configurePayloadSigning(AwsSigningConfig signingConfig, bool
     }
 
     private static SignedRequest doSign(SignRequest<? extends AwsCredentialsIdentity> request,
+                                        Checksummer checksummer,
                                         AwsSigningConfig signingConfig,
                                         V4aPayloadSigner payloadSigner) {
+        SdkHttpRequest.Builder requestBuilder = request.request().toBuilder();
         ContentStreamProvider contentStreamProvider = request.payload().orElse(null);
+
+        checksummer.checksum(contentStreamProvider, requestBuilder);
+        payloadSigner.beforeSigning(requestBuilder, contentStreamProvider, signingConfig.getSignedBodyValue());
```

Comment:
just double checking, could moving `beforeSigning` call ahead for anonymous request cause any issue?

Reply:
It seems anonymous requests are not supported in sigv4a, so the ordering doesn't really matter, but I moved  beforeSigning and checksum methods call down to preserve the existing behavior